purchases/src/main/kotlin/com/revenuecat/purchases/virtualcurrencies/VirtualCurrencyManager.kt

```diff
@@ -0,0 +1,94 @@
+package com.revenuecat.purchases.virtualcurrencies
+
+import com.revenuecat.purchases.common.AppConfig
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.caching.DeviceCache
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.interfaces.GetVirtualCurrenciesCallback
+
+@Suppress("UnusedPrivateProperty")
+internal class VirtualCurrencyManager(
+    private val identityManager: IdentityManager,
+    private val deviceCache: DeviceCache,
+    private val backend: Backend,
+    private val appConfig: AppConfig,
+) {
+
+    // region Public functions
+
+    @Suppress("ForbiddenComment")
+    @Synchronized
+    fun virtualCurrencies(
+        callback: GetVirtualCurrenciesCallback,
+    ) {
+        val appUserID = identityManager.currentAppUserID
+        val isAppBackgrounded = appConfig.isAppBackgrounded
+
+        val cachedVirtualCurrencies = fetchCachedVirtualCurrencies(
+            appUserID = appUserID,
+            isAppBackgrounded = isAppBackgrounded,
+            allowStaleCache = false,
+        )
+        if (cachedVirtualCurrencies != null) {
+            callback.onReceived(cachedVirtualCurrencies)
+            return
+        }
+
+        fetchVirtualCurrenciesFromBackend(
+            appUserID = appUserID,
+            isAppBackgrounded = isAppBackgrounded,
+            callback = callback,
+        )
+
+        // TODO: Cache the VCs from the network
```

Comment:
I know the fetch from network method is not implemented, but it seems like we could already call `cacheVirtualCurrencies` here? is there something else missing?

Reply:
All that's missing is the network call that gives us the value to provide to `cacheVirtualCurrencies()`. I was going to add it here, but this gave a compilation error since `virtualCurrenciesFromNetwork` isn't initialized:

```kotlin
val virtualCurrenciesFromNetwork: VirtualCurrencies // TODO: Fetch from network
        cacheVirtualCurrencies(
            virtualCurrencies = virtualCurrenciesFromNetwork,
            appUserID = appUserID
        )
```

Reply:
Added this here: 9ac9904e813e51cc88ebfba1ed615e76008771cb